### ISSUE TITLE
Sending data back to django does not work after upgrade to 0.5.x

### ISSUE THREAD

**Issue by jsvh** [jorgebastida/django-dajaxice#69]
I trying to update my page using server-side mini-templates and 'injecting' it into the html using javascript.  Using latest trunk.

I send the following:

Dajaxice.my_app.some_function(injection, 
{
                    'json_string': $(this).parents('form').to_json(this.name),
            'origin': this.id,
                    'destination': 'show-select',
                    'type': 'html'
                }, 
{'error_callback': custom_error});

Now i have a function in ajax.py

def some_function(request, json_string, origin, destination, type):
.... does some stuff

Now I am getting the following error:

Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/django-trunk/django/core/handlers/base.py", line 116, in get_response response = callback(request, _callback_args, *_callback_kwargs)
  File "/Library/Python/2.7/site-packages/django-trunk/django/views/generic/base.py", line 64, in view return self.dispatch(request, _args, *_kwargs)
  File "/Library/Python/2.7/site-packages/django-dajaxice/dajaxice/views.py", line 52, in dispatch response = function.call(request, *_data)
  File "/Library/Python/2.7/site-packages/django-dajaxice/dajaxice/core/Dajaxice.py", line 18, in call return self.function(_args, **kwargs)
TypeError: some_function() takes exactly 5 arguments (1 given)

For some reason the javascript function is not passing the argument variables over to django.  Not really sure why.  Hopefully someone can help.

**Comment by jsvh** [comment]
Does not work in the development-trunk but does work in 1.4.2

**Comment by mrginglymus** [comment]
Hi, I have this problem too - it seems to come from this change: 

https://github.com/django/django/commit/dfd4a7175119ddb422d8426dcc15902265d5a428

specifically,

> Non-form data in HTTP requests
> 
> :attr:`request.POST <django.http.HttpRequest.POST>` will no longer include data posted via HTTP requests with non form-specific content-types in the header. In prior versions, data posted with content-types other than `multipart/form-data` or `application/x-www-form-urlencoded` would still end up represented in the :attr:`request.POST <django.http.HttpRequest.POST>` attribute. Developers wishing to access the raw POST data for these cases, should use the :attr:`request.body <django.http.HttpRequest.body>` attribute instead.

**Comment by mrginglymus** [comment]
The monkeypatch in views.py is
`data = unquote(request.body.split('=')[1])`
until something more solid can be done

**Comment by lduchesne** [comment]
In case someone needs a quick patch (python 3.3, django 1.5):

```
--- views.py    2012-12-27 13:49:51.000000000 -0500
+++ views.py    2012-12-27 14:03:43.000000000 -0500
@@ -8,6 +8,8 @@
 from dajaxice.exceptions import FunctionNotCallableError
 from dajaxice.core import dajaxice_functions, dajaxice_config

+from urllib.parse import unquote
+
 log = logging.getLogger('dajaxice')


@@ -17,7 +19,7 @@
     http://www.gossamer-threads.com/lists/python/bugs/684379
     """
     if isinstance(d, dict):
-        return dict([(k.encode('utf-8'), safe_dict(v)) for k, v in d.iteritems()])
+        return dict([(k, safe_dict(v)) for k, v in d.items()])
     elif isinstance(d, list):
         return [safe_dict(x) for x in d]
     else:
@@ -36,7 +38,7 @@
         if dajaxice_functions.is_callable(name, request.method):

             function = dajaxice_functions.get(name)
-            data = getattr(request, function.method).get('argv', '')
+            data = unquote(request.body.decode('utf-8').split('=')[1])

             # Clean the argv
             if data != 'undefined':
```

**Comment by horejsek** [comment]
I fixed it [here](https://github.com/horejsek/django-dajaxice/commit/c23085e7b91117bec42114048ee71ca771b7f8d2).
[Pull request](https://github.com/jorgebastida/django-dajaxice/pull/75) is waiting.

**Comment by jorgebastida** [comment]
Merged.
It's going to be released as 0.5.5 soon
